server/src/main/java/org/elasticsearch/common/hash/MurmurHash3.java

```diff
@@ -65,34 +65,35 @@ protected static long fmix(long k) {
      * @param length - length of array
      * @return - hashcode
      */
-    public static long hash64(byte[] data, int length) {
-        return hash64(data, length, DEFAULT_SEED);
+    public static long hash64(byte[] data, int length, int offset) {
```

Comment:
Maybe swap the arguments.  Most methods that work with bytes and support offset/length  have the offset parameter before the length parameter

Reply:
Swapped here and in further method with seed as well.